xla/service/collective_ops_utils.cc

```diff
@@ -449,6 +449,67 @@ StatusOr<std::vector<GlobalDeviceId>> GetParticipatingDevices(
   }
 }
 
+StatusOr<std::vector<int64_t>> GetPariticipantCountsForReplicaGroups(
+    int64_t num_replicas, int64_t num_partitions,
+    absl::Span<const ReplicaGroup> replica_groups,
+    CollectiveOpGroupMode group_mode) {
+  std::vector<int64_t> participant_counts;
+  std::vector<ReplicaGroup> participating_replica_groups =
+      SpanToVector(replica_groups);
+
+  // If replica groups are empty, assume a group with all replicas.
+  if (replica_groups.empty()) {
+    if (group_mode == CollectiveOpGroupMode::kFlattenedID) {
+      // replica groups contain flattened-ids and cannot be empty.
+      TF_RET_CHECK(!replica_groups.empty())
+          << "replica groups cannot be empty for kFlattenedID mode";
+    }
+
+    int total_participant_count;
+    if (group_mode == CollectiveOpGroupMode::kCrossPartition) {
+      // replica group are partition ids.
+      total_participant_count = num_partitions;
+    } else {
+      // replica group are replica ids.
+      total_participant_count = num_replicas;
+    }
+
+    ReplicaGroup replica_group = ReplicaGroup();
+    for (int id = 0; id < total_participant_count; id++) {
+      replica_group.add_replica_ids(id);
+    }
+    participating_replica_groups.push_back(replica_group);
+  }
+
+  switch (group_mode) {
+    case CollectiveOpGroupMode::kCrossReplica: {
+      for (const auto& replica_group : participating_replica_groups) {
```

Comment:
unused `replica_group` is the complaint here from internal tests

Reply:
updated this to remove replica_group